### ISSUE TITLE
German support for follow next/previous links

### ISSUE THREAD

**Issue by bovine3dom** [cmcaine/tridactyl#324]
Courtesy of Lexi on Riot.

> # followpagepatterns.prev = ^(prev(ious)?|older|vorherig.*|älter.*|zurück)\b|»|>>
> # followpagepatterns.next = ^(next|newer|neuer.*|nächste.*|weiter.*)\b|»|>>
> Not sure whether the .*s will cause problems, it was just faster then putting in all of neuer, neuere, neuerer, neueres, neueren, etc for all the grammatical cases.

**Comment by cmcaine** [comment]
This already works. Lexi's issue is something else.

**Comment by bovine3dom** [comment]
No, we should bake it in, right?

**Comment by cmcaine** [comment]
Yeah, we could detect the language of the document and switch patterns or
just leave them in there all the time.

Don't know what the false positive rate would be.

**Comment by glacambre** [comment]
Looks like we can check HTTP Headers, attributes on the HTML tag and HTML meta tags in order to determine the language of a page: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Language .
But I doubt this is very used in the wild. Still, it's probably better than nothing.

I don't have any idea about how to handle properly handle internationalization but I know there are a few EcmaScript APIs for this: https://hacks.mozilla.org/2014/12/introducing-the-javascript-internationalization-api/ .

Once an architecture is decided upon I can translate Tridactyl in French if needed.

**Comment by bovine3dom** [comment]
I reckon we can just have all languages coexisting peacefully.

**Comment by bovine3dom** [comment]
This is a really easy issue to investigate if anyone wants to get started. Just play with `set followpagepatterns.next` and `set followpagepatterns.prev`, adding as many languages as you like, and see if there are any regressions on common pages. Then add the settings you chose to `src/lib/config.ts`.
